### PR TITLE
Add version check of conda and update package versions

### LINKED ISSUE #142: Update installation instructions to use conda 23.10.0 or later
Previously we modified our installation instructions to tell people how to install the libmamba library to make the conda installation go faster. As of [conda version 23.10.0](https://conda.org/blog/2023-11-06-conda-23-10-0-release/), libmamba is the default solver.

For the Astropy workshop installation, we are [modifying the instructions](https://github.com/astropy/astropy-workshop/pull/225) to tell people to update their conda versions and check that they have version 23.10.0 or later. We should do the same for our instructions.

### PR DESCRIPTION
**Pull request by stargaser**
* Add instructions for users to check that their conda version is at least 23.10.0 that includes the libmamba solver by default
* Advise users to update their conda version if needed
* Remove the steps to install and use libmamba-solver
* Update the Python version and package versions

Fixes #142

### CODE REVIEW

File: check_env.py [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
-        'numpy': ('1.24', None),
-        'matplotlib': ('3.2', None),
-        'jupyterlab': ('3.1', None),
-        'astropy': ('5.2', None),
-        'scipy': ('1.0', None),
-        'pyvo': ('1.4', None),
+        'numpy': ('1.26', None),
+        'matplotlib': ('3.8', None),
+        'jupyterlab': ('4.0', None),
+        'astropy': ('6.0', None),
```
Comment (bsipocz) [inline check_env.py:18]: Not sure whether we can do this yet, I think we need to disentangle this version check from the conda environment versions. 

E.g. the notebooks should work with older astropy, no need to get the latest version (especially that a lot of packages in the ecosystem is still incompatible with astropy6).

So I would keep this list for the minimum versions, we do check for these in CI anyway, and the conda env separate.

File: environment.yml [hunk 2]
```diff
@@ -2,13 +2,13 @@ name: navo-env
...
-  - astropy>=5.2
-  - matplotlib>=3.2
-  - jupyterlab>=3.1
-  - pyvo>=1.4
-  - scipy>=1.0
+  - python=3.11
+  - numpy>=1.26
+  - astropy>=6.0
+  - matplotlib>=3.8
+  - jupyterlab>=4.0
```
Comment (bsipocz) [inline environment.yml:9]: Heads up: There was an incompatibility with the md notebooks with jupyter-server 2+, so we need to keep an eye on it that none of these brings in that package as a dependency (or if they do, they pick up the correct version)

File: check_env.py (hunk 1 above)
Comment (stargaser) [reply check_env.py:18]: @bsipocz so are you recommending to set the environment check to, say, version 5.3, and leave the `environment.yml` entry for astropy as-is?

Comment (bsipocz) [reply check_env.py:18]: or even leave 5.2 and we bump versions when we run into limitations (or when they are getting a bit out of date).

Totally separately you can update the versions in `environment.yml` as that is a combination users are expected to setup for the workshop itself (in my mind, the notebooks should be able to run separately from that specific environment, and indeed we test with the minimum versions separatetly)

Comment (bsipocz) [reply check_env.py]: I'm sorry I wasn't clear. This comment is true for all the minimum versions, bump them when needed or when SPEC0 or similar would say they are outdated, but don't require latest and greatest by default.

**Review by tomdonaldson — APPROVED** [review]
Let's get this in for integration with next CI PR.

**Pull request merged**